vllm/model_executor/sampling_metadata.py

```diff
@@ -466,22 +465,30 @@ def from_lists(cls, temperatures: List[float], top_ps: List[float],
         do_penalties = prompt_tokens or output_tokens
 
         if do_penalties:
```

Comment:
I have merged the if-else blocks based on `do_penalties` together. Not sure why they were separated in the first place.